test_conformance/conversions/test_conversions.cpp

```diff
@@ -1141,421 +1210,444 @@ static int DoTest( cl_device_id device, Type outType, Type inType, SaturationMod
+        if ((error = clEnqueueWriteBuffer(gQueue, gInBuffer, CL_TRUE, 0,
+                                          count * gTypeSizes[inType], gIn, 0,
+                                          NULL, &writeInputBuffer)))
         {
-            vlog_error( "ERROR: clEnqueueWriteBuffer failed. (%d)\n", error );
+            vlog_error("ERROR: clEnqueueWriteBuffer failed. (%d)\n", error);
             gFailCount++;
             goto exit;
         }
 
-        // Setup completion callback for the write, which will enqueue the rest of the work
-        // This is somewhat gratuitous.  Because this is an in order queue, we didn't really need to
-        // do this work in a callback. We could have done it from the main thread.  Here we are
-        // verifying that the implementation can enqueue work from a callback, while at the same time
-        // also checking to make sure that the conversions work.
+        // Setup completion callback for the write, which will enqueue the rest
+        // of the work This is somewhat gratuitous.  Because this is an in order
+        // queue, we didn't really need to do this work in a callback. We could
+        // have done it from the main thread.  Here we are verifying that the
+        // implementation can enqueue work from a callback, while at the same
+        // time also checking to make sure that the conversions work.
         //
-        // Because the verification code is also moved to a callback, it is hoped that implementations will
-        // achieve a test performance improvement because they can verify the results in parallel.  If the
-        // implementation serializes callbacks however, that won't happen.   Consider it some motivation
-        // to do the right thing! :-)
-        if( (error = clSetEventCallback( writeInputBuffer, CL_COMPLETE, WriteInputBufferComplete, &writeInputBufferInfo)) )
-        {
-            vlog_error( "ERROR: clSetEventCallback failed. (%d)\n", error );
-            gFailCount++;
-            goto exit;
-        }
-
-        // The event can't be destroyed until the callback is called, so we can release it now.
-        if( (error = clReleaseEvent(writeInputBuffer) ))
-        {
-            vlog_error( "ERROR: clReleaseEvent failed. (%d)\n", error );
-            gFailCount++;
-            goto exit;
-        }
+        // Because the verification code is also moved to a callback, it is
+        // hoped that implementations will achieve a test performance
+        // improvement because they can verify the results in parallel.  If the
+        // implementation serializes callbacks however, that won't happen.
+        // Consider it some motivation to do the right thing! :-)
+        WriteInputBufferComplete((void *)&writeInputBufferInfo);
 
         // Make sure the work is actually running, so we don't deadlock
-        if( (error = clFlush( gQueue ) ) )
+        if ((error = clFlush(gQueue)))
```

Comment:
Minor: I don't think this flush is needed anymore either, since we've already issued a blocking write on this queue.